### PR TITLE
Implement ResponseAccessor

### PR DESCRIPTION
**Pull request by xepozz**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌
| Issues  | #4

### CODE REVIEW

File: src/ResponseAccessor.php [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+    {
+        return $this->response->getProtocolVersion();
+    }
+
+    /**
+     * @param string $version
+     * @return ResponseAccessor
+     * @psalm-suppress LessSpecificReturnStatement
+     */
+    public function withProtocolVersion($version): ResponseAccessor
```
Comment (samdark) [inline src/ResponseAccessor.php:39]: What do you need it for in the tests?

File: src/ResponseAccessor.php [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+        return $this->response->getHeaderLine($name);
+    }
+
+    /**
+     * @param string $name
+     * @param string|string[] $value
+     * @return ResponseAccessor
+     * @psalm-suppress LessSpecificReturnStatement
+     */
+    public function withHeader($name, $value): ResponseAccessor
```
Comment (samdark) [inline src/ResponseAccessor.php:71]: What do you need it for in tests?

File: src/ResponseAccessor.php [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+        return new self($response);
+    }
+
+    /**
+     * @param string $name
+     * @param string|string[] $value
+     * @return ResponseAccessor
+     * @psalm-suppress LessSpecificReturnStatement
+     */
+    public function withAddedHeader($name, $value): ResponseAccessor
```
Comment (samdark) [inline src/ResponseAccessor.php:83]: What do you need it for in tests?

File: src/ResponseAccessor.php [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+        $response = $this->response->withAddedHeader($name, $value);
+        return new self($response);
+    }
+
+    /**
+     * @param string $name
+     * @return ResponseAccessor
+     * @psalm-suppress LessSpecificReturnStatement
+     */
+    public function withoutHeader($name): ResponseAccessor
```
Comment (samdark) [inline src/ResponseAccessor.php:94]: What do you need it for in tests?

File: src/ResponseAccessor.php [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+    {
+        return $this->response->getBody();
+    }
+
+    /**
+     * @param StreamInterface $body
+     * @return ResponseAccessor
+     * @psalm-suppress LessSpecificReturnStatement
+     */
+    public function withBody(StreamInterface $body): ResponseAccessor
```
Comment (samdark) [inline src/ResponseAccessor.php:110]: What do you need it for in tests?

File: src/ResponseAccessor.php [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+        return $this->response->getStatusCode();
+    }
+
+    /**
+     * @param int $code
+     * @param string $reasonPhrase
+     * @return ResponseAccessor
+     * @psalm-suppress LessSpecificReturnStatement
+     */
+    public function withStatus($code, $reasonPhrase = ''): ResponseAccessor
```
Comment (samdark) [inline src/ResponseAccessor.php:127]: What do you need it for in tests?

**Comment by xepozz** [comment]
That tests immutability

**Pull request merged**